tests/unit/heartbeat_tests.py

```diff
@@ -20,15 +20,39 @@
 # pylint: disable=C0103
 
 
+class ConstructableConnection(connection.Connection):
+    """Adds dummy overrides for `Connection`'s abstract methods so
+    that we can instantiate and test it.
+
+    """
+    def _adapter_connect_stream(self):
+        pass
+
+    def _adapter_disconnect_stream(self):
+        raise NotImplementedError
+
+    def add_timeout(self, deadline, callback):
+        raise NotImplementedError
+
+    def remove_timeout(self, timeout_id):
+        raise NotImplementedError
+
+    def _adapter_emit_data(self, data):
+        raise NotImplementedError
+
+    def _adapter_get_write_buffer_size(self):
+        raise NotImplementedError
+
+
 class HeartbeatTests(unittest.TestCase):
```

Comment:
@michaelklishin - The heartbeat unit tests had logic to test this, but the mock spec was not sufficiently detailed to allow the access to the bogus attribute `heartbeat` to be detected.